plugins/modules/vault_kv2_write.py

```diff
@@ -0,0 +1,221 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+# (c) 2023, Devon Mar (@devon-mar)
+# GNU General Public License v3.0+ (see LICENSES/GPL-3.0-or-later.txt or https://www.gnu.org/licenses/gpl-3.0.txt)
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
+
+DOCUMENTATION = """
+module: vault_kv2_write
+version_added: 4.2.0
+author:
+  - Devon Mar (@devon-mar)
+short_description: Perform a write operation against a KVv2 secret in HashiCorp Vault
+description:
+  - Perform a write/patch operation against a KVv2 secret in HashiCorp Vault.
+requirements:
+  - C(hvac) (L(Python library,https://hvac.readthedocs.io/en/stable/overview.html))
+  - For detailed requirements, see R(the collection requirements page,ansible_collections.community.hashi_vault.docsite.user_guide.requirements).
+extends_documentation_fragment:
+  - community.hashi_vault.attributes
+  - community.hashi_vault.attributes.action_group
+  - community.hashi_vault.connection
+  - community.hashi_vault.auth
+  - community.hashi_vault.engine_mount
+attributes:
+  check_mode:
+    support: partial
+    details:
+      - If I(read) is C(true), full check mode functionality is supported.
+      - If I(read) is C(false), the status will always be C(changed) but a write will not be performed in check mode.
+options:
+  engine_mount_point:
+    type: str
+    default: secret
+  path:
+    type: str
+    required: true
+    description:
+      - Vault KVv2 path to be written to.
+      - This is relative to the I(engine_mount_point), so the mount path should not be included.
+  data:
+    type: dict
+    required: true
+    description:
+      - KVv2 secret data to write/patch.
+  cas:
+    type: int
+    description:
+      - Perform a check-and-set operation.
+  read:
+    type: bool
+    default: false
+    description:
+      - Read the secret first and write only when I(data) differs from the read data.
+      - Requires C(read) permission on the secret if C(true).
+      - If C(false), this module will always write to I(path).
+"""
+
+EXAMPLES = r"""
+- name: Write/create a secret
+  community.hashi_vault.vault_kv2_write:
+    url: https://vault:8201
+    path: hello
+    data:
+      foo: bar
+
+- name: Write with check and set
+  community.hashi_vault.vault_kv2_write:
+    url: https://vault:8201
+    path: caspath
+    cas: 0
+    data:
+      foo: bar
```

Comment:
Let's add an example where `cas` is not `0`, and another where it is `0`, labeled (via `name`) such that it explains that `0` means it will only succeed if the secret does not exist.